sources/api/netdog/src/cli/write_resolv_conf.rs

```diff
@@ -0,0 +1,49 @@
+use super::{error, Result};
+use crate::dns::DnsSettings;
+use crate::{LEASE_FOLDER, PRIMARY_INTERFACE};
+use argh::FromArgs;
+use snafu::ResultExt;
+use std::fs;
+use std::path::{Path, PathBuf};
+
+#[derive(FromArgs, PartialEq, Debug)]
+#[argh(subcommand, name = "write-resolv-conf")]
+/// Writes /etc/resolv.conf, using DNS API settings if they exist
+pub(crate) struct WriteResolvConfArgs {}
+
+pub(crate) fn run() -> Result<()> {
+    // Use DNS API settings if they exist, otherwise use the primary interface's lease
+    let dns_settings =
+        if let Some(settings) = DnsSettings::from_config().context(error::DnsFromConfSnafu)? {
+            settings
+        } else {
+            let lease_path = primary_interface_lease()?;
+            DnsSettings::from_lease_file(lease_path).context(error::DnsFromLeaseSnafu)?
+        };
+    dns_settings
+        .write_resolv_conf()
+        .context(error::ResolvConfWriteFailedSnafu)?;
+    Ok(())
+}
```

Comment:
Can the merging of the DNS config settings and the lease settings be a helper function? Especially if becomes slightly more complicated than using one source or the other.

We're doing almost but not quite exactly the same thing as in the DHCP helper entry point, which seems like it could lead to bugs.